examples/nrf24l01_btle/nrf24l01_btle.c

```diff
@@ -0,0 +1,635 @@
+          outbuf[16], outbuf[17], outbuf[18], outbuf[19], outbuf[20],
+          outbuf[21], outbuf[22], outbuf[23]);
+  syslog(LOG_INFO,
+         "0x%02x,0x%02x,0x%02x,0x%02x,0x%02x,0x%02x,0x%02x,0x%02x\n",
+          outbuf[24], outbuf[25], outbuf[26], outbuf[27], outbuf[28],
+          outbuf[29], outbuf[30], outbuf[31]);
+}
+#endif
+
+int nrf24_cfg(int fd)
+{
+  int error = 0;
+
+  uint32_t rf = NRF24L01_MIN_FREQ + frequency[current];
+  int32_t txpow = DEFAULT_TXPOWER;
+
+  /**
+   * if using RATE_1Mbps from include/nuttx/wireless/nrf24l01.h,
+   * tools/checkpatch.sh report error: Mixed case identifier found.
+   **/
+
+  nrf24l01_datarate_t datarate = 0; /* RATE_1Mbps */
+  nrf24l01_retrcfg_t retrcfg =
+    {
+      .count = 0,
+      .delay = 3 /* DELAY_1000us */
+    };
+
+  uint32_t addrwidth = 4;
+
+  uint8_t pipes_en = (1 << 0);  /* Only pipe #0 is enabled */
+
+  /**
+   * Define the pipe #0 parameters (AA enabled and dynamic payload length)
+   * the address must be 0x8e89bed6.
+   **/
+
+  nrf24l01_pipecfg_t pipe0cfg =
+    {
+     .en_aa = false,
+     .payload_length = 32,
+     .rx_addr =
+        {
+          swapbits(0x8e), swapbits(0x89), swapbits(0xbe), swapbits(0xd6)
+        }
+    };
+
+  nrf24l01_pipecfg_t *pipes_cfg[NRF24L01_PIPE_COUNT] =
+    {
+      &pipe0cfg, 0, 0, 0, 0, 0
+    };
+
+  nrf24l01_state_t primrxstate;
+
+#ifdef CONFIG_WL_NRF24L01_RXSUPPORT
+  primrxstate = ST_RX;
+#else
+  primrxstate = ST_POWER_DOWN;
+#endif
+
+  /* Set radio parameters */
+
+  ioctl(fd, NRF24L01IOC_SETRETRCFG,
+        (unsigned long)((nrf24l01_retrcfg_t *)&retrcfg));
+
+  ioctl(fd, WLIOC_SETRADIOFREQ, (unsigned long)((uint32_t *)&rf));
+  ioctl(fd, WLIOC_SETTXPOWER, (unsigned long)((int32_t *)&txpow));
+  ioctl(fd, NRF24L01IOC_SETDATARATE,
+        (unsigned long)((nrf24l01_datarate_t *)&datarate));
+
+  ioctl(fd, NRF24L01IOC_SETADDRWIDTH,
+        (unsigned long)((uint32_t *)&addrwidth));
+
+  /* set advertisement address: 0x8E89BED6 (bit-reversed -> 0x6B7D9171) */
+
+  ioctl(fd, NRF24L01IOC_SETTXADDR,
+           (unsigned long)((uint8_t *)&pipe0cfg.rx_addr));
+
+  ioctl(fd, NRF24L01IOC_SETPIPESCFG,
+        (unsigned long)((nrf24l01_pipecfg_t **)&pipes_cfg));
+  ioctl(fd, NRF24L01IOC_SETPIPESENABLED,
+        (unsigned long)((uint8_t *)&pipes_en));
+
+  ioctl(fd, NRF24L01IOC_SETSTATE,
+        (unsigned long)((nrf24l01_state_t *)&primrxstate));
+
+  return error;
+}
+
+int nrf24_open(void)
+{
+  int fd;
+
+  fd = open(DEV_NAME, O_RDWR);
+
+  if (fd < 0)
+    {
+      perror("Cannot open nRF24L01 device");
+    }
+  else
+    {
+      nrf24_cfg(fd);
+    }
+
+  return fd;
+}
+
+int nrf24_send(int wl_fd, uint8_t * buf, uint8_t len)
+{
+  int ret;
+
+  wlinfo("send buffer len %d\n", len);
+  ret = write(wl_fd, buf, len);
+  if (ret < 0)
+    {
+      wlerr("Error sending packet\n");
+      return ret;
+    }
+
+  return OK;
+}
+
+static inline void random_mac(void)
```

Comment:
why not just call libc `arc4random_buf` to generate the random bytes.

Reply:
Hi, @btashton . I have did test of your suggestion, If using libc `arc4random_buf ` , it dependency  enable `CONFIG_CRYPTO_RANDOM_POOL`.  if enable ` CONFIG_CRYPTO_RANDOM_POOL` and `CONFIG_DEBUG_FEATURES` I got report that `arm-none-eabi-ld: region `flash' overflowed by 1104 bytes` on stm32f103-minimum. Also, I  call libc `arc4random_buf` to generate random MAC, this app not working, which means it not discovered  by BLE scanner.

Reply:
Yeah I guess that is kind of heavy.  You could also iterate over calls `random()` which should be very cheap and will give you a pseudo random number.

Reply:
OK, I added a new option by generate random, give user a choice, maybe the random mac  not suitable for some few people.